processor/src/main/java/org/mapstruct/ap/internal/model/IterableMappingMethod.java

```diff
@@ -379,6 +385,15 @@ else if ( other.selectionParameters != null ) {
             return false;
         }
 
+        if ( this.factoryMethod != null ) {
```

Comment:
Can you use `if (!Objects.equals(this.factoryMethod, other.factoryMethod))`? I know that we already have the same `if`s for the `selectionParameters`, but I find it much clearer when it is one liner with `Objects.equals`.